ws/ws.nim

```diff
@@ -183,6 +183,45 @@ proc connect*(
     onClose = onClose,
     rng = rng)
 
+proc selectExt(ws: WSServer, request: HttpRequest): string
+  {.raises: [Defect, WSExtError].} =
+
+  # it is possible to have multiple "Sec-WebSocket-Extensions"
+  let exts = request.headers.getList("Sec-WebSocket-Extensions")
+  var extList: seq[AppExt]
```

Comment:
I think `AppExt` is the same as `Ext`, should we merge them?

Reply:
I agree ExtParam should be moved to `types.nim`.

My idea to separate `AppExt` and `Ext` is to separate server code(ext creation) and parser code. If we want to merge them, then the parser code and server code will be interleaving.  But I will revisit this later to find a clean solution.

Reply:
I think that for the most part, the client/server functionality is very similar, if we can merge them I'd prefer we do that.